Cannot lint .vue files, even after attempting many suggested changes and the glob pattern in the tsconfig targets vue files

**Repro**
I ran into an issue that was brought up here a LOT (#1303 among others). No matter what I tried, I couldn't get the parser to recognize the `.vue` file. I've tried everything suggested in different issues, a separate `tsconfig.eslint.json`, different glob patterns, none worked...

The repository is available at https://github.com/GAwesomeBot/web-next. All you need to do is `npm i` and then `npm run lint` (add `--debug` to run with debug). The other team member also encountered the same issues.

We always get the same error
```
> @gawesomebot/web@1.0.0 lint C:\Users\Vlad\Desktop\Development\GAwesomeBot Next\web
> eslint --ext .vue,.ts . "--debug"


C:\Users\Vlad\Desktop\Development\GAwesomeBot Next\web\src\components\test.vue
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: src\components\test.vue.
The file must be included in at least one of the projects provided

1 problem (1 error, 0 warnings)
```

Per the documentation, as you can see in our `.eslintrc.json` file in the repository, we have
```json
	"parser": "vue-eslint-parser",
	"parserOptions": {
		"parser": "@typescript-eslint/parser",
		"project": "./tsconfig.eslint.json",
		"ecmaFeatures": {
			"jsx": true,
			"modules": true
		},
		"sourceType": "module",
		"extraFileExtensions": [".vue"]
	},
```

as well as both tsconfig files including `.vue` files

```json
  
{
	"extends": "./tsconfig.json",
	"include": [
		"**/*.ts",
		"**/*.vue"
	],
	"exclude": [
		"node_modules/"
	]
}
```

Am I doing something wrong? Did I miss a step? If so, I'm sorry! 😓 

**Expected Result**

The `.vue` file(s) would get picked up and parsed accordingly.

**Actual Result**

The `.vue` file(s) were not picked up.

**Additional Info**

Skimming through the `--debug` logs, I got this https://hasteb.in/udogojak.yaml, which is the error log for linting the vue file.
I used the stable version of the parser before, I tried `canary` thinking it might solve the issue. 

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `2.25.1-alpha.4` |
| `TypeScript`                | `3.9.0-dev.20200326` |
| `ESLint`                    | `6.8.0` |
| `node`                      | `12.14.1` |
| `npm`                       | `6.14.3` |

Running on Windows 10 x64 1909 (build 18363.720)

--- a/packages/typescript-estree/src/create-program/WatchCompilerHostOfConfigFile.ts
+++ b/packages/typescript-estree/src/create-program/WatchCompilerHostOfConfigFile.ts
@@ -32,6 +32,7 @@ interface WatchCompilerHostOfConfigFile<T extends ts.BuilderProgram>
   onCachedDirectoryStructureHostCreate(
     host: CachedDirectoryStructureHost,
   ): void;
+  extraFileExtensions?: readonly ts.FileExtensionInfo[];
 }
 
 export { WatchCompilerHostOfConfigFile };

--- a/packages/typescript-estree/src/create-program/createWatchProgram.ts
+++ b/packages/typescript-estree/src/create-program/createWatchProgram.ts
@@ -315,8 +315,12 @@ function createWatchProgram(
     oldOnDirectoryStructureHostCreate(host);
   };
   // This works only on 3.9
-  (watchCompilerHost as any).extraFileExtensions = extra.extraFileExtensions.map(
-    extension => ({ extension, scriptKind: ts.ScriptKind.Deferred }),
+  watchCompilerHost.extraFileExtensions = extra.extraFileExtensions.map(
+    extension => ({
+      extension,
+      isMixedContent: true,
+      scriptKind: ts.ScriptKind.Deferred,
+    }),
   );
   watchCompilerHost.trace = log;
 

--- a/packages/typescript-estree/src/create-program/createWatchProgram.ts
+++ b/packages/typescript-estree/src/create-program/createWatchProgram.ts
@@ -314,6 +314,10 @@ function createWatchProgram(
       );
     oldOnDirectoryStructureHostCreate(host);
   };
+  // This works only on 3.9
+  (watchCompilerHost as any).extraFileExtensions = extra.extraFileExtensions.map(
+    extension => ({ extension, scriptKind: ts.ScriptKind.Deferred }),
+  );
   watchCompilerHost.trace = log;
 
   // Since we don't want to asynchronously update program we want to disable timeout methods

